src/Build.UnitTests/Microsoft.Build.Engine.UnitTests.csproj

```diff
@@ -43,7 +43,7 @@
     <ProjectReference Include="..\Samples\ProjectCachePlugin\ProjectCachePlugin.csproj" Private="false" ReferenceOutputAssembly="false">
       <SetTargetFramework Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework'">TargetFramework=$(FullFrameworkTFM)</SetTargetFramework>
       <SetTargetFramework Condition="'$(TargetFrameworkIdentifier)' == '.NETFramework' and '$(MonoBuild)' == 'true'">TargetFramework=$(FullFrameworkTFM)</SetTargetFramework>
-      <SetTargetFramework Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">TargetFramework=netcoreapp2.1</SetTargetFramework>
+      <SetTargetFramework Condition="'$(TargetFrameworkIdentifier)' != '.NETFramework'">TargetFramework=net5.0</SetTargetFramework>
     </ProjectReference>
```

Comment:
Is this project reference really needed? It appears to build and run fine without it.

Reply:
Somehow you want to specify that the sample project should be built before the test project. The test project does not have a compile time dependency on it, but a runtime dependency, because the tests are testing whether an arbitrary, previously unknown plugin assembly can be found and its types loaded and instantiated. I had hit issues with this dependency ordering while changing both test and sample, and this ProjectReference is the only way I found to make it work.